include/mbedtls/rsa.h

```diff
@@ -69,10 +69,6 @@
 extern "C" {
 #endif
 
-#if !defined(MBEDTLS_RSA_ALT)
```

Comment:
Regarding the removal of `MBEDTLS_RSA_ALT`, we should also merge rsa_alt_helpers.c into rsa.c with an additional `MBEDTLS_STATIC_TESTABLE` qualifier to all functions and merge rsa_alt_helpers.h in rsa_internal.h with the function prototypes guarded by  `MBEDTLS_TEST_HOOKS`. See for example the definition and declaration of `mbedtls_ssl_tls13_parse_certificate()` respectively in ssl_tls13_generic.c and ssl_tls13_invasive.h.

Reply:
> we should also merge rsa_alt_helpers.c into rsa.c

Should we? Would we gain significant code size, especially in a public-key only configuration? (Most use cases for small code size also avoid RSA private-key operations because they require a lot of CPU time. Code size does matter somewhat for signature verification, which is commonly used in bootloaders due to the quick verification and small code size.)

Keeping the file separation would facilitate future backports.

Reply:
I hadn't in mind code size gain. The documentation in rsa_alt_helpers.h states that functions in rsa_alt_helpers.c are intended to be used in RSA alternative implementations and as we remove this possibility, the documented reason for rsa_alt_helpers does not hold anymore. Thus it made sense to me to merge it back to rsa.c, did not think about future backports though. Would rsa_alt_helpers.c be beneficial for PSA drivers? At least, we should adapt the documentation in rsa_alt_helpers.h.

Reply:
> rsa_alt_helpers.c be beneficial for PSA drivers?

That's doubtful: the code that is most likely to be useful to PSA drivers is in `rsa.c` and often only parts of functions. It's in fact likely that a lot of code from `rsa_alt_helpers.c` will disappear due to https://github.com/Mbed-TLS/TF-PSA-Crypto/issues/105 . I suggest waiting until we have a decision on that.

Reply:
> I suggest waiting until we have a decision on that.

Fine by me. @tom-daubney-arm please just update the documentation in rsa_alt_helpers.h then.

Reply:
I have modified the documentation through writing a note at the top explaining that the docs are out of date and that the file is due for a refactor. PTAL.

Reply:
This looks good to me thanks. Following the merge of #9299 you need to rebase though and unfortunately you need to rebase also the other ALT-related PRs.